.github/workflows/prepare-transaction.yml

```diff
@@ -0,0 +1,207 @@
+              exit 1
+            fi
+          fi
+
+      - name: Check the chain in chainlist
+        if: success() && ${{ steps.validate_description.outputs.chain_id != null }}
+        env:
+          CHAIN_ID: ${{ steps.validate_description.outputs.chain_id }}
+        run: |
+          trim() {
+              # 3.5.3 Shell Parameter Expansion at https://www.gnu.org/software/bash/manual/bash.html
+              local var="$*"
+              # remove leading whitespace characters
+              var="${var#"${var%%[![:space:]]*}"}"
+              # remove trailing whitespace characters
+              var="${var%"${var##*[![:space:]]}"}"
+              echo "$var"
+          }
+
+          echo "$CHAIN_ID from the RPC URL is valid. Checking the chain in chainlist..."
+          chainlist_url="https://raw.githubusercontent.com/ethereum-lists/chains/master/_data/chains/eip155-$CHAIN_ID.json"
+
+          chainlist_status_code=$(trim "$(curl -LI $chainlist_url -o /dev/null -w '%{http_code}\n' -s)")
+
+          if [ "$chainlist_status_code" == "404" ]; then
+            echo "Chain $CHAIN_ID is not in chainlist."
+            echo "COMMENT_OUTPUT="$ERROR_MSG_CHAINLIST_FAILURE"" >> $GITHUB_ENV
+            exit 1
+          fi
+
+      - name: Estimate Transaction and Check Pre-fund
+        if: success() && ${{ steps.validate_description.outputs.rpc_url != null }}
+        env:
+          RPC_URL: ${{ steps.validate_description.outputs.rpc_url }}
+        run: |
+          json_request='[
+              {
+                  "jsonrpc": "2.0",
+                  "method": "eth_gasPrice",
+                  "params": [],
+                  "id": 1
+              },
+              {
+                  "jsonrpc": "2.0",
+                  "method": "eth_estimateGas",
+                  "params": [
+                      {
+                          "from": "'$FACTORY_DEPLOYER_ADDRESS'",
+                          "data": "'$FACTORY_BYTECODE'"
+                      }
+                  ],
+                  "id": 2
+              }
+          ]'
+
+          response=$(curl "$RPC_URL" --location --header 'Content-Type: application/json' --data "$json_request")
+
+          if jq -e . >/dev/null 2>&1 <<< "$response"; then
+            gas_price=$(jq -r '.[0].result' <<< "$response")
+            echo "Current gas price: $gas_price"
+            gas_limit=$(jq -r '.[1].result' <<< "$response")
+            echo "Estimated deployment gas limit: $gas_limit"
+
+            if [[ "$gas_price" == "null" || "$gas_limit" == "null" ]]; then
+              echo "COMMENT_OUTPUT="$ERROR_MSG_PREFUND_CHECK"" >> $GITHUB_ENV
+              exit 1
+            fi
+
+            # We multiply the gas limit by 1.4, just like the deployment script does it
+            gas_limit=$(( $gas_limit * 14 / 10 ))
+            expected_prefund=$(($gas_limit * $gas_price))
```

Comment:
Do these need to have the `0x` stripped like for chainId ☝️?

Reply:
I tested it, and it looks like Bash is capable of arithmetic in hex and decimal, and the result is in decimal.

Example:
```
echo $((0xbeef * 14/10))
68430
```